### PR TITLE
replaces Brew with supported installer for AWS CLI 

### PR DESCRIPTION
**Pull request by brennanMKE**
https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-mac.html

*Issue #, if available:*

#3777

*Description of changes:*

Changes command to install AWS CLI from brew to the supported process for macOS.

*Check points:*

- [x] Added new tests to cover change, if needed
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Updated CHANGELOG.md
- [x] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by royjit — APPROVED** [review]
LGTM, @palpatim any concerns?

File: .circleci/config.yml [hunk 1]
```diff
@@ -137,9 +137,10 @@ commands:
       Install aws cli and configure android aws release profile
```
Comment (palpatim) [inline .circleci/config.yml:137]:
```suggestion
      Install aws cli and configure release profile
```

**Review by palpatim — APPROVED** [review]
minor comment change

**Review by lawmicha — APPROVED** [review]

**Comment by lawmicha** [comment]
I was able to test this from this https://github.com/aws-amplify/aws-sdk-ios/pull/3796 with workflow https://app.circleci.com/pipelines/github/aws-amplify/aws-sdk-ios/5138/workflows/5a87200b-4e6b-4ba3-8a78-d93bfce24a33/jobs/82590

**Pull request merged**